### ISSUE TITLE
wandb.ai not logging epochs vs metrics/losses instead uses step

### ISSUE THREAD

**Issue by lrbrrr** [ultralytics/yolov5#2175]
Dataset: coco128

`!git clone https://github.com/ultralytics/yolov5  # clone repo
%cd yolov5
%pip install -qr requirements.txt  # install dependencies

import torch
from IPython.display import Image, clear_output  # to display images

clear_output()
print('Setup complete. Using torch %s %s' % (torch.__version__, torch.cuda.get_device_properties(0) if torch.cuda.is_available() else 'CPU'))

torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip', 'tmp.zip')
!unzip -q tmp.zip -d ../ && rm tmp.zip

%pip install -q wandb  
!wandb login  # use 'wandb disabled' or 'wandb enabled' to disable or enable

!python train.py --batch 16 --epochs 20 --data coco128.yaml --weights yolov5s.pt --name test --cache

`
Output: x-axis is step and y-axis is metrics/losses

Expected: x-axis as epochs
![Screenshot from 2021-02-09 12-52-57](https://user-images.githubusercontent.com/78729788/107484969-26d3ec80-6ba9-11eb-9dac-bf942e07a8fa.png)

**Comment by glenn-jocher** [comment]
@lrbrrr hmm ok thanks, let me see if I can reproduce!

**Comment by glenn-jocher** [comment]
@lrbrrr if I open up a new colab notebook and run Setup cell, and then wandb install and login, and then train (COCO128 auto-downloads on first use) using your command for 20 epochs, then my W&B console looks like this, with the x axis going from 0 to 19 steps (I also ran the default 3 epoch training), so 1 step = 1 epoch. Are you seeing something different? I can't see your image because you included it in the code block.

<img width="1000" alt="Screen Shot 2021-02-10 at 12 51 23 AM" src="https://user-images.githubusercontent.com/26833433/107486625-1df80200-6b3a-11eb-907e-cec4b70875cf.png">

**Comment by lrbrrr** [comment]
It's strange that I see the results as yours for coco128 dataset now but not for custom dataset training.

For custom dataset, I see this result. Trained fresh an hour ago
COCO128 trained for 100 epochs
![image](https://user-images.githubusercontent.com/78729788/107487913-b3cc7500-6bac-11eb-86c2-a79213076f22.png)
CUSTOM trained for 100 epochs
![image](https://user-images.githubusercontent.com/78729788/107488063-deb6c900-6bac-11eb-88bd-f3b7ef7998fa.png)

**Comment by glenn-jocher** [comment]
@lrbrrr oh that is really strange! The epochs update was very recent, only last week, so if you are training the custom models on an older repo this can happen.

I'm training several models on full COCO now and the correct epochs appear on x axis.

**Comment by lrbrrr** [comment]
It does work for COCO. 

For custom data: I've encountered new issue now

I've organized the directory as per [this tutorial](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data).

![Screenshot from 2021-02-11 10-39-29](https://user-images.githubusercontent.com/78729788/107604350-e41c1e00-6c55-11eb-8bd8-c85bff4df377.png)

**Comment by glenn-jocher** [comment]
@lrbrrr it can't find your labels. Please follow the instructions in the custom training tutorial, and match the coco128 or coco formats if possible:
https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data

**Comment by lrbrrr** [comment]
I've organized the directory as mentioned in the tutorial and labels are present too with correct format.

**Comment by lrbrrr** [comment]
> It does work for COCO. Also I'm using the current repo using git clone on colab.
> 
> For custom data: I've encountered new issue now
> 
> I've organized the directory as per [this tutorial](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data).
> 
> ![Screenshot from 2021-02-11 10-39-29](https://user-images.githubusercontent.com/78729788/107604350-e41c1e00-6c55-11eb-8bd8-c85bff4df377.png)

This error was generated because I used images with metadata included. After cleaning it worked fine. It said INVALID IMAGE FORMAT MPO. Preprocessing the images worked smoothly. 

I can see epoch vs metrics/losses.

Thank you for the help

**Comment by Jelly123456** [comment]
@lrbrrr how do you solve the "Invalid image format MPO" error? I am also facing this error now.

**Comment by lrbrrr** [comment]
@Jelly123456 Probably the images that you are using right now comes with metadata. I guess the model expects images with pixel information only. 

I tried opencv-python to read the pixel information and rewrite it to new_image_name.jpg
That's it!

**Comment by lrbrrr** [comment]
```
for i in range(len(os.listdir(directory))):
    file = os.listdir(directory)[i]
    img = cv2.imread(os.path.join(directory, file))
    name, ext = os.path.splitext(file)
    new_name = name + '.jpg'
    save_path = to_dir + new_name
    cv2.imwrite(save_path, img)
    print(f'{file} saved to {save_path}.', end='\r')

```

**Comment by 93yh** [comment]
谢谢您！以前的版本可以的丫，为啥4.0的带元数据（无人机拍的照片）的就不可以了呢？这是负优化？

**Comment by thesauravs** [comment]
@glenn-jocher might answer how and why it worked in older and not with the recent release.

**Comment by glenn-jocher** [comment]
@sahil20021008, that's an interesting comparison between the results for the COCO and custom dataset trainings. The images you have shared indicate that while the metrics are displayed for COCO, they are not being displayed for the custom dataset training.

It seems like you are not seeing the metrics for the custom dataset training on the W&B website. The metrics can be logged using `--log-imgs all` argument along with the `--project` argument in the `train.py` script.  
Ensure that you are logging the metrics correctly using this argument and that you're currently configured to use W&B for logging metrics.

Let me know if this resolves the issue.
